modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/capabilities/CapabilitiesServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.runtime.capabilities;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.nuxeo.common.Environment.DISTRIBUTION_HOTFIX;
+import static org.nuxeo.common.Environment.DISTRIBUTION_NAME;
+import static org.nuxeo.common.Environment.DISTRIBUTION_SERVER;
+import static org.nuxeo.common.Environment.DISTRIBUTION_VERSION;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.ComponentContext;
+import org.nuxeo.runtime.model.ComponentManager;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * @since 11.5
+ */
+public class CapabilitiesServiceImpl extends DefaultComponent implements CapabilitiesService {
```

Comment:
This probably needs a `getApplicationStartedOrder()` with a very low value to be sure that all other services requiring it are started afterwards.

Reply:
This is not strictly required in this case as the `start` method execution is not required to make the mechanism working.

But it is still a good practice and doesn't cost much, I will add it.